wiring/src/spark_wiring_ble.cpp

```diff
@@ -1728,15 +1733,23 @@ ssize_t BleLocalDevice::getAdvertisingData(BleAdvertisingData* advertisingData)
     if (advertisingData == nullptr) {
         return SYSTEM_ERROR_INVALID_ARGUMENT;
     }
-    return hal_ble_gap_get_advertising_data(advertisingData->data(), BLE_MAX_ADV_DATA_LEN, nullptr);
+    advertisingData->clear();
+    ssize_t len = hal_ble_gap_get_advertising_data(advertisingData->data(), BLE_MAX_ADV_DATA_LEN, nullptr);
```

Comment:
Prefer
```cpp
size_t len = CHECK(hal_ble_gap_get_advertising_data(advertisingData->data(), BLE_MAX_ADV_DATA_LEN, nullptr));
```

Same in https://github.com/particle-iot/device-os/pull/1891/files#diff-93f9dd9012bb15fde13e60440775ab25R1749